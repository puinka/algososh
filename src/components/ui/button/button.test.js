import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("should render Button component", () => {
  it("with text", () => {
    const button = renderer.create(<Button text={"text"} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("with no text", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("as disabled", () => {
    const button = renderer.create(<Button disabled />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("as loading", () => {
    const button = renderer.create(<Button isLoader />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("fire a click event", () => {
    const mockCallBack = jest.fn();

    render(<Button text={"buttonText"} onClick={mockCallBack} />);

    const button = screen.getByText("buttonText");

    fireEvent.click(button);

    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
