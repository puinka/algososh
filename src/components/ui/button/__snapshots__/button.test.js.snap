// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component as disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`should render Button component as loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`should render Button component with no text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`should render Button component with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
